src/web/js-interop.md

```diff
@@ -22,3 +22,77 @@ For help using the `js` package, see the following:
 [js-api]: {{site.pub-api}}/js
 [sass]: {{site.pub-pkg}}/sass
 
+## Future-proof JS interop
+
+Dart's JS interop story is currently evolving. 
+Many of the features that enable future JS interop,
+and support of all three compilers,
+are ready to use (Dart version 2.19). 
+To ease the transition to future JS interop,
+you can refactor your code to conform to the
+syntax and semantics of the new features now.
+
+{{site.alert.note}}
+  Future-proofing is for those *very* eager to
+  prepare their code for [Dart 3][], or to try
+  integrating with [`dart2wasm`][] before Dart 3.
+  None of this necessarily works *performantly* on
+  `dart2wasm` yet, but refactoring now means you
+  won't need to later, if that's important to you.
+{{site.alert.end}}
+
+The following sections are the set of features
+expected to work across compilers for JS interop,
+per library.
+
+*Requirements:*
+* Dart SDK constraint: `>= 2.19`
+* [`package:js`][] constraint: `>= 0.6.6`
+
+[Dart 3]: https://medium.com/dartlang/the-road-to-dart-3-afdd580fbefa
+[`dart2wasm`]: https://github.com/dart-lang/sdk/blob/main/pkg/dart2wasm/dart2wasm.md#running-dart2wasm
+[`package:js`]: {{site.pub-pkg}}/js
+
+### `package:js`
+
+If you're interested in interop that will work with
+`dart2wasm` when Dart 3 is released, you can implement
```

Comment:
It's too early to state confidently that `dart2wasm` will 100% be in Dart 3.

Reply:
Also - I am not sure how much emphasis we are putting on dart2wasm on other documentation. I see that we are staring to mention it in the blog post, but I wasn't sure how comfortable we are emphasizing it here.

Circling back to the sentence here. My take is that "Next generation interop" addresses several gaps we had in JSInterop in the past:

* missing features: before we couldn't easily wrap and transform APIs, rename members, and we missed static checking. With static interop we can now do all of these.
* inconsistencies: before development vs production web compilers sometimes behaved differently. With static interop this is now better, backends are more consistent. **
* clarity: JSInterop adds cognitive challenges because it is using Dart syntax in a non-Dart way. Static interop doesn't solve this yet, but inline-classes will help us make this feature be more idiomatic.

** Since our investment in dart2wasm, we discovered that details above like the lack of static checking made existing JSInterop approaches prohibitively expensive to support in a non JavaScript environment (like wasm). Static interop enables us to have an interop solution that we can support better in dart2wasm, and as a result will enable experimentation on that backend whenever it is available as a preview.

Reply:
Great, I'll focus the section on addressing the general improvements instead of directly relating it to dar2wasm